docs/querying/api.md

```diff
@@ -695,6 +695,8 @@ URL query parameters:
 - `file[]=<string>`: only return rules with the given filepath. If the parameter is repeated, rules with any of the provided filepaths are returned. When the parameter is absent or empty, no filtering is done.
 - `exclude_alerts=<bool>`: only return rules, do not return active alerts. 
 - `match[]=<label_selector>`: only return rules that have configured labels that satisfy the label selectors. If the parameter is repeated, rules that match any of the sets of label selectors are returned. Note that matching is on the labels in the definition of each rule, not on the values after template expansion (for alerting rules). Optional.
+- `max_groups=<number>`: return up to `max_groups` rule groups. In case there are more groups, a `nextToken` property will be present in the response. The value in that property can be used in subsequent requests in the `next_token` property to paginate over the remaining rule groups. The `nextToken` property will not be present in the last response.
```

Comment:
Can we add mention about no guarantee from API that pagination to be "atomic" / "transactional" or "consistent" (whatever we want to call it)? As-is there is no guarantee (for now) that pagination will NOT miss certain rules or even fail (the token referencing group that was just removed) in the event of changes in the duration of pagination?

For the fail on token=group not found reason, would simple offset on number of groups, be a simpler approach? 🤔

Reply:
I will update the documentation.

> For the fail on token=group not found reason, would simple offset on number of groups, be a simpler approach?
I agree that it is simpler, but the issue with this approach is that it allow users to make assumptions about how the api works under the hood, removing flexibility from us to change the implementation in the future. the token should be meaningless due to this reason.